test/controllers/applications_controller_test.rb

```diff
@@ -47,6 +47,28 @@ class ApplicationsControllerTest < ActionController::TestCase
     end
   end
 
+  describe '#new' do
+    it 'redirects to the event if the application process is run by the organizer' do
+      event = make_event(application_process: 'application_by_organizer',
+                         application_link: 'http://www.something.org')
+
+      post :new,
+        event_id: event.id
+
+      assert_redirected_to event
+    end
+
+    it 'adds a new application if the selection process is not run by organizer' do
+      event = make_event
+
+      post :new,
+        event_id: event.id
+
+      assert_response :success
```

Comment:
Count +1 of Application.all to make sure a new event has been created?

assert_equal 1, Application.all.count